File: DocumentAnalysisSystem/Models/Paragraph.cs
using DocumentAnalysisSystem.Visitors;

namespace DocumentAnalysisSystem.Models
{
    public class Paragraph : DocumentComponent
    {
        public Paragraph(string text) : base("Paragraph")
        {
            Text = text;
        }

        public string Text { get; }

        public override IReadOnlyList<DocumentComponent> Children => new DocumentComponent[0];

        public override void Accept(IDocumentVisitor visitor)
        {
            visitor.Visit(this);
        }
    }
}

File: DocumentAnalysisSystemJava/src/main/java/com/documentanalysis/models/Paragraph.java
package com.documentanalysis.models;

import com.documentanalysis.visitors.IDocumentVisitor;

import java.util.Collections;
import java.util.List;

public class Paragraph extends DocumentComponent {
    private final String text;

    public Paragraph(String text) {
        super("Paragraph");
        this.text = text;
    }

    public String getText() {
        return text;
    }

    @Override
    public List<DocumentComponent> getChildren() {
        return Collections.emptyList();
    }

    @Override
    public void accept(IDocumentVisitor visitor) {
        visitor.visit(this);
    }
}
